src/Utilities/MoistThermodynamics/MoistThermodynamics.jl

```diff
@@ -681,6 +681,9 @@ function saturation_adjustment(e_int::DT, ρ::DT, q_tot::DT) where DT
     T, converged = find_zero(
       T -> internal_energy_sat(T, ρ, q_tot) - e_int,
       T_1, T_2, SecantMethod(); xatol=DT(1e-3), maxiters=10)
+      if !converged
+        @show "SA did not converge" #TODO - error/warning handling
```

Comment:
Instead of ``SA``, spell out ``saturation_adjustment.`` (Better yet: we can assure convergence with proper initial brackets and regula falsi.)

Reply:
My only hesitation with using Regula Falsi method is that it has trouble running on the GPU (because there are several `if` statements, which creates branches, and GPUs are not good with handling many branches). Maybe @simonbyrne and @vchuravy can weigh in on this?

It may also be nice if we can map out the 3D space for which we believe `saturation_adjustment` should yield valid results. If the inputs are not physically realistic, do we still want to return a value without any warning/error? This is something we can do once and then specify valid ranges in the docs.

Reply:
Good point about GPUs. Probably even with the secant method we should get convergence in reasonable cases as long as the initial bracket is chosen appropriately.

Reply:
Yes. My question was more about what we want to do in unreasonable cases. For example when the simulation becomes unstable because of the instability in rain. I would expect the saturation adjustment scheme to throw some error or warning when it doesn't converge.

Reply:
Yes, error handling is good and important. We should just make sure that this isn't triggered in usual circumstances.